R/utils.R

```diff
@@ -34,3 +34,23 @@ calc_weights <- function(net) {
     sfnetworks::activate("edges") |>
     dplyr::mutate(weight = sfnetworks::edge_length())
 }
+
+#' Get UTM zone from longitude
+#'
+#' @param x An sf object
+#'
+#' @return The UTM zone
+#' @export
+get_utm_zone_epsg <- function(x) {
+  if (!"sf" %in% class(x)) {
+    stop("x must be an sf object")
+  }
+
+  long <- x |>
+    sf::st_transform(4326) |>
+    sf::st_union() |>
+    sf::st_centroid() |>
+    sf::st_coordinates()
+
+  32600 + floor((long[1] + 180) / 6) + 1
```

Comment:
I think this formula is valid only for the Northern emisphere, for the Southern emishpere it should be `32700 + ...` (suggested to rename `long` to `coords` in the previous comment as the latitude also becomes relevant):

```suggestion
  if (coords[2] >= 0.){
    base <- 32600
  } else {
    base <- 32700
  }
  
  base + floor((coords[1] + 180) / 6) + 1
```